workers/unity/Packages/io.improbable.gdk.testutils/MockWorld.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Subscriptions;
+using Unity.Entities;
+using UnityEngine;
+
+namespace Improbable.Gdk.TestUtils
+{
+    public class MockWorld : IDisposable
+    {
+        public struct Options
+        {
+            public string WorkerType;
+            public Type[] AdditionalSystems;
+            public ILogDispatcher Logger;
+        }
+
+        public WorkerInWorld Worker { get; private set; }
+        public MockConnectionHandler Connection { get; private set; }
+
+        public EntityGameObjectLinker Linker { get; private set; }
+
+        private HashSet<GameObject> gameObjects = new HashSet<GameObject>();
+
+        public static MockWorld Create(Options options)
+        {
+            var mockWorld = new MockWorld();
+
+            var connectionBuilder = new MockConnectionHandlerBuilder();
+            mockWorld.Connection = connectionBuilder.ConnectionHandler;
+
+            mockWorld.Worker = WorkerInWorld
+                .CreateWorkerInWorldAsync(connectionBuilder,
+                    options.WorkerType ?? "TestWorkerType",
+                    options.Logger ?? new LoggingDispatcher(),
+                    Vector3.zero)
+                .Result;
+
+            foreach (var type in options.AdditionalSystems ?? new Type[] { })
+            {
+                mockWorld.Worker.World.CreateSystem(type);
+            }
+
+            mockWorld.Linker = new EntityGameObjectLinker(mockWorld.Worker.World);
+
+            PlayerLoopUtils.ResolveSystemGroups(mockWorld.Worker.World);
+
+            return mockWorld;
+        }
+
+        public T GetSystem<T>() where T : ComponentSystemBase
+        {
+            return Worker.World.GetExistingSystem<T>();
+        }
+
+        public MockWorldWithContext<T> Step<T>(Func<MockWorld, T> frame)
+        {
+            var context = frame(this);
+            Update();
+            return new MockWorldWithContext<T>
+            {
+                Context = context,
+                World = this
+            };
+        }
+
+        public MockWorld Step(Action<MockWorld> frame)
+        {
+            frame(this);
+            Update();
+            return this;
+        }
+
+        private void Update()
+        {
+            Worker.World.Update();
+        }
+
+        public (GameObject, T) CreateGameObject<T>(long entityId) where T : MonoBehaviour
+        {
+            var gameObject = new GameObject("TestGameObject");
+            gameObjects.Add(gameObject);
+
+            var component = gameObject.AddComponent<T>();
+            component.enabled = false;
```

Comment:
I think the component will have already called `OnEnabled` at this point.

If it has, you might want to instantiate the test object as a child to a disabled root game object, to avoid it waking up until setup is complete.

Reply:
Had a look at this and paired offline - it seems that in Editmode tests `OnEnable` isn't called